components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/API.java

```diff
@@ -904,6 +912,69 @@ public String getEndpointConfig() {
         if ((endpointConfig == null || StringUtils.isAllEmpty(endpointConfig) && endpoints.size() > 0)) {
             return getEndpointConfigString(endpoints);
         }
+
+        if (isEndpointSecured()) {
+            try {
+                JSONParser parser = new JSONParser();
+                ObjectMapper objectMapper = new ObjectMapper();
+                JSONObject endpointConfigJson = (JSONObject) parser.parse(endpointConfig);
+                EndpointSecurity productionAndSandbox = new EndpointSecurity();
+                if (endpointConfigJson.get(APIConstants.ENDPOINT_SECURITY) == null) {
+                    JSONObject epSecurity = new JSONObject();
+                    productionAndSandbox.setEnabled(true);
+                    if (isEndpointAuthDigest()) {
+                        productionAndSandbox.setType(APIConstants.ENDPOINT_SECURITY_TYPE_DIGEST);
+                    } else {
+                        productionAndSandbox.setType(APIConstants.ENDPOINT_SECURITY_TYPE_BASIC);
+                    }
+                    productionAndSandbox.setUsername(getEndpointUTUsername());
+                    productionAndSandbox.setPassword(getEndpointUTPassword());
+                    Object productionAndSandboxSecurity = parser.parse(
+                            objectMapper.writeValueAsString(productionAndSandbox));
+                    epSecurity.put(APIConstants.ENDPOINT_SECURITY_PRODUCTION, productionAndSandboxSecurity);
+                    epSecurity.put(APIConstants.ENDPOINT_SECURITY_SANDBOX, productionAndSandboxSecurity);
+                    endpointConfigJson.put(APIConstants.ENDPOINT_SECURITY, epSecurity);
```

Comment:
This seems to be repetitive code throughout if else code blocks. Can we get this out of the if condition?

Reply:
These if-else represent three separate scenarios and the object the value set is being varied as per the scenario.